components/service-operator/internal/aws/sdk/client.go

```diff
@@ -54,10 +56,13 @@ type client struct {
 	*ecr.ECR
 }
 
-func (c *client) AssumeRole(roleARN string) Client {
+func (c *client) GetRoleCredentials(roleARN string) *credentials.Credentials {
 	sess := session.Must(session.NewSessionWithOptions(session.Options{
 		SharedConfigState: session.SharedConfigEnable,
 	}))
-	creds := stscreds.NewCredentials(sess, roleARN)
-	return NewClient(&aws.Config{Credentials: creds})
+	return stscreds.NewCredentials(sess, roleARN)
+}
+
+func (c *client) AssumeRole(roleARN string) Client {
```

Comment:
Do we need to be able to set the session duration here? ... I think the default is 1hr, but our controller is only gonna come back every 6hrs (since that's what the docker creds ttl is)

https://godoc.org/github.com/aws/aws-sdk-go/service/sts#AssumeRoleInput

Reply:
think you're right

Reply:
https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_GetAuthorizationToken.html says 12 but ok